src/newsblur_api.cpp

```diff
@@ -10,55 +8,14 @@
 #include <newsblur_api.h>
 #include <string.h>
 #include <time.h>
-#include <password.h>
 
 #define NEWSBLUR_ITEMS_PER_PAGE 6
 
 namespace newsbeuter {
 
 newsblur_api::newsblur_api(configcontainer * c) : remote_api(c) {
-	std::string user = cfg->get_configvalue("newsblur-login");
-	bool flushed = false;
-
-	if (user == "") {
-		std::cout << std::endl;
-		std::cout.flush();
-		flushed = true;
-		std::cout << "Username for Newsblur: ";
-		std::cin >> user;
-
-		if (user.empty())
-			LOG(level::CRITICAL, "newsblur_api::newsblur_api: No user set");
-	}
-
-	std::string pass = cfg->get_configvalue("newsblur-password");
-	std::string pass_file = cfg->get_configvalue("newsblur-passwordfile");
-	std::string pass_eval = cfg->get_configvalue("newsblur-passwordeval");
-	if ( pass == "" ) {
-		if ( pass_file != "" ) {
-			pass = read_password(pass_file);
-		} else if ( pass_eval != "") {
-			pass = eval_password(pass_eval);
-		} else {
-			if (!flushed) {
-				std::cout << std::endl;
-				std::cout.flush();
-			}
-			// Find a way to do this in C++ by removing cin echoing.
-			pass = std::string( getpass("Password for Newsblur: ") );
-		}
-
-		if (pass.empty())
-			LOG(level::CRITICAL, "newsblur_api::newsblur_api: No password set");
-	}
-
-	auth_info = strprintf::fmt("username=%s&password=%s", user, pass);
 	api_location = cfg->get_configvalue("newsblur-url");
 	min_pages = (cfg->get_configvalue_as_int("newsblur-min-items") + (NEWSBLUR_ITEMS_PER_PAGE + 1)) / NEWSBLUR_ITEMS_PER_PAGE;
-
-	if (cfg->get_configvalue("cookie-cache").empty()) {
```

Comment:
Why move these lines out of constructor? If you wanted to prevent authentication when `cookie-cache` isn't set, you should've filed this as an issue and work on it separately.